mopidy/core/playback.py

```diff
@@ -14,21 +14,25 @@
 class PlaybackController(object):
     pykka_traversable = True
 
-    def __init__(self, backends, core):
+    def __init__(self, audio, backends, core):
+        # TODO: these should be internal
         self.backends = backends
         self.core = core
+        self._audio = audio
 
-        self._current_tl_track = None
         self._stream_title = None
         self._state = PlaybackState.STOPPED
 
-    def _get_backend(self):
-        # TODO: take in track instead
-        track = self.get_current_track()
-        if track is None:
+        self._current_tl_track = None
+        self._pending_tl_track = None
+
+        if self._audio:
+            self._audio.set_about_to_finish_callback(self._on_about_to_finish)
```

Comment:
The call path should be something like the following:

```
about-to-finish --> audio --> core callback --> backend --> audio 
```

The goal here is to block the about to finish until an `audio.set_uri` type call has made it out. (This is also one of the reasons I want backends out of this path). The `about-to-finish` should be coming from some gst internal thread, so our audio code will also run there, and then from there core would probably still be in the same context to how the callback is done. We den call into the backend via actors, after which the backend calls into audio via actors. So we must ensure that we can't deadlock that last call, or else the `about-to-finish` never returns.